client/src/features/Campaign/StepDialog.js

```diff
@@ -9,64 +9,196 @@ import {
-                  className={classes.select} >
-            {
-            props.templates.map( (template) => {
-              return (
-                <MenuItem key={template.id} value={template.id}>{template.name}</MenuItem>
-              )
-            })
-            }
-          </Select>
-        </FormControl>
+        <Grid container
+              alignItems="center"
+              justify="space-around" >
+          <Grid item xs={12} sm={9}>
+            <FormControl className={classes.fullWidth} variant="filled">
+              <InputLabel id="template-select-label">Template</InputLabel>
+              <Select labelId="template-select-label"
+                      id="template-select"
+                      value= {props.step ? props.step.templateId : null}
+                      onChange={e => props.step ? setTemplate(e) : props.setTemplateId(e.target.value)} 
+                      onChange={handleChange}
+                      className={classes.select} >
+                {
+                props.templates.map( (template) => {
+                  return (
+                    <MenuItem key={template.id} value={template.id}>{template.name}</MenuItem>
+                  )
+                })
+                }
+              </Select>
+            </FormControl>
+          </Grid>
+          <Grid item xs={12} sm={3}>
+            <Button variant="contained" 
+                    size="small" 
+                    color="primary" 
+                    onClick={newTemplate}
+                    className={classes.newButton} >
+              New
+            </Button>
+          </Grid>
+        </Grid>
+        <TemplateEditor template={editorTemplate}
+                        onSave={handleSaveTemplate} />
       </DialogContent>
       <DialogActions>
         { props.delete ? <Button onClick={props.onDeleteClick} color="secondary" className={classes.deleteButton}>Delete</Button> : null }
-        <Button onClick={props.onSave} color="primary">
-          Save
+        <Button onClick={handleSave} color="primary">
+          Save Step
         </Button>
-        <Button onClick={props.onClose} color="default">
+        <Button onClick={handleClose} color="default">
           Cancel
         </Button>
       </DialogActions>
+
+      <Snackbar
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left',
+        }}
+        open={snackbarOpen}
+        autoHideDuration={3000}
+        onClose={handleSnackbarClose}
+        ContentProps={{
+          'aria-describedby': 'message-id',
+        }}
+        message={<span id="message-id">Template saved</span>}
+        action={[
+          <IconButton
+            key="close"
+            aria-label="close"
+            color="inherit"
+            className={classes.close}
+            onClick={handleSnackbarClose}
+          >
+            <CloseIcon />
+          </IconButton>,
```

Comment:
There's a SuccessSnackbar in the features, can probably refine it to handle that